ballerina/http_client.bal

```diff
@@ -31,7 +31,7 @@ client isolated class HttpClient {
         return createSimpleHttpClient(self, globalHttpClientConnPool, url, options, options.toString());
     }
 
-    # The `HttpClient.post()` function can be used to send HTTP POST requests to HTTP endpoints.
+    # Create new data or submit information to an HTTP endpoint.
```

Comment:
The HTTP spec defines the terminology `resource` to explain the methods. A POST method requests the target resource to process the data enclosed in the request.

In the new descriptions, it seems like we are inconsistently using the terms resource and data. Better to align with resource term.

```suggestion
    # Create a new resource on the endpoint or submit data to a resource for processing
```